GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -1210,14 +1210,9 @@ private void UpdatePlaceholders()
                 folderPlaceholdersToKeep.Add(string.Empty);
 
                 stopwatch.Restart();
-                this.ProcessListOnThreads(
-                    numThreads,
-                    placeholderFilesListCopy,
-                    (placeholderBatch, start, end, blobSizesConnection, availableSizes) =>
-                        this.BatchPopulateMissingSizesFromRemote(blobSizesConnection, placeholderBatch, start, end, availableSizes),
-                    (placeholder, blobSizesConnection, availableSizes) =>
-                        this.UpdateOrDeleteFilePlaceholder(blobSizesConnection, placeholder, folderPlaceholdersToKeep, availableSizes));
+                this.MultiThreadedPlaceholderUpdatesAndDeletes(numThreads, placeholderFilesListCopy, folderPlaceholdersToKeep);

```

Comment:
We could move the `numThreads` calculation into this method since that is the only place it is used.